### PR TITLE
differential fuzzing

### PR DESCRIPTION
**Pull request by marctrem**
This PR introduces support for differential fuzzing.

* new `make shared` target for building Industry-compatible targets `$(call make-shared,...)` in Local.mk
* new `industry-bundle` target for building "$OBJDIR/industry-bundle.zip"

### CODE REVIEW

File: config/with-industry.mk
[the remark this answers is not in the document]
Comment (marctrem) [reply config/with-industry.mk:1]: I've just changed this config to import the `with-fuzz.mk` configuration. That said, we need another config to build a shared object (and generate position-independent code).

Please let me know if you'd rather have it done another way.

File: contrib/build_industry_targets.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  make -j industry-test
+
+for fd_target in $(find ./build/linux/clang/icelake/industry-test -type f -executable); do
+  NAME=$(basename $fd_target)
+  mkdir -p "$INDUSTRY_TARGETS/$NAME"
+  cp "$fd_target" "$INDUSTRY_TARGETS/$NAME/firedancer.so"
+done
+
+pushd ffi/rust
+cargo build
```
[the remark this answers is not in the document]
Comment (marctrem) [reply contrib/build_industry_targets.sh:25]: This is not the long-term solution. This is just to get unblocked. I'd rather not spend too much time on the temporary solution (unless you have something nicer to suggest).

File: config/with-shared.mk
[the remark this answers is not in the document]
Comment (marctrem) [reply config/with-shared.mk:1]: All objects need to be PIC so it needs to be project-wide.

File: src/ballet/sbpf/industry_sbpf_loader.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  *res_out = -1;
+  *out_result_sz = sizeof(int32_t);
+}
+
+int
+industry_test_one( unsigned long * out_result_sz,
+                   unsigned char * out_result_buf,
+                   unsigned long   result_buf_sz,
+                   unsigned char * data,
+                   unsigned long   data_sz ) {
```
[the remark this answers is not in the document]
Comment (marctrem) [reply src/ballet/sbpf/industry_sbpf_loader.c:60]: Behaviours can be set at init through argv (industry_init).
If it's a per-test behaviour, then, data can be used.

Isn't this sufficient?

[the remark this answers is not in the document]
Comment (marctrem) [reply src/ballet/sbpf/industry_sbpf_loader.c:60]: We could totally add a void ptr arg in the future if it helps.

**Comment by ripatel-fd** [comment]
@marctrem Thanks your your contribution. As discussed, I'll build on top of your work adding build system support and expose targets via shared objects. I'll also include a guide on how to build these objects. The shared objects will provide the project industry APIs and experimental strongly typed APIs.

**Pull request closed without merging**